src/ore/src/metrics/delete_on_drop.rs

```diff
@@ -164,14 +223,21 @@ where
     L: PromLabelsExt<'a>,
 {
     fn from_metric_vector(vec: &GenericCounterVec<P>, labels: L) -> Self {
-        let inner = labels.get_from_metric_vec(vec);
+        let inner = Some(labels.get_from_metric_vec(vec));
         Self {
             inner,
             labels,
             vec: vec.clone(),
             _phantom: &PhantomData,
         }
     }
+
+    /// Leaks the inner counter. Use with caution!
+    ///
+    /// Prevents dropping the metric label.
```

Comment:
I think this is a really risky function to call, especially on metrics generated from user-controlled strings: If things go wrong, you'll attempt to leak a metric with the same label set as another metric that remains droppable, then the second metric gets dropped & deleted from the registry. I'm not sure I know if prometheus will let you keep reporting the first metric (or if it'll panic on, e.g., `inc()`)

Reply:
If we do need it, I feel like this might be a candidate to be declared `unsafe`

Reply:
Yep, I agree! I removed it from the PR.